File: 1822-sign-of-the-product-of-an-array/1822-sign-of-the-product-of-an-array.py
class Solution:
    def arraySign(self, nums: List[int]) -> int:
        zero = False
        negatives = 0
        for i in nums:
            if i < 0:
                negatives += 1
            if i == 0:
                zero = True
        if zero:
            return 0
        if negatives % 2 == 1:
            return -1
        return 1
        

File: 1822-sign-of-the-product-of-an-array/1822-sign-of-the-product-of-an-array.java
class Solution {
    public int arraySign(int[] nums) {
        int negativeNumbers = 0;
        
        for (int num : nums) {
            if (num == 0) {
                return 0;
            }
            if (num < 0) {
                negativeNumbers += 1;
            }
        }
        return negativeNumbers % 2 == 0 ? 1 : -1;
        
    }
}